### PR TITLE
Select optimizer in model-builder

### PR DESCRIPTION
**Pull request by Lewuathe**
Since now two optimizers ("sgd", "momentum") are available, optimizer can be also option to be set in model builder.
screen shot 2017-09-10 at 12 38 01

---
This change is [Reviewable](https://reviewable.io/reviews/pair-code/deeplearnjs/114)

### CODE REVIEW

**Comment by nsthorat** [comment]
This is awesome, thank you!

Can you wait until this PR goes in and we can add 2 more optimizers? https://github.com/PAIR-code/deeplearnjs/pull/102

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-:-Ktdw1v2S0Ak1L-Bl63H:bymudh5)*

**Comment by nsthorat** [comment]
Oh one more thing, I pushed the model builder:
https://pair-code.github.io/deeplearnjs/demos/model-builder/model-builder-demo.html

Can you in this PR add default values to the hyperparams?



---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-:-KtdxyEjG43D1gQZE5kl:b-z7qiu2)*

**Comment by Lewuathe** [comment]
Sure I'll wait #102 and update accordingly.

> Can you in this PR add default values to the hyperparams?

https://pair-code.github.io/deeplearnjs/demos/model-builder/model-builder-demo.html
has already have default values for learning rate, momentum and batch size. What can I do?

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-:-KtekpQOqSockTzprqVX:b-6pdigc)*

**Comment by nsthorat** [comment]
The other PR is merged! You can add those to the dropdown if you'd like! :)

**Comment by Lewuathe** [comment]
@nsthorat Thanks. I'll update the patch.

**Comment by Lewuathe** [comment]
I added Adagrad and RMSProp to model builder.

screen shot 2017-09-18 at 17 40 01

**Comment by nsthorat** [comment]
Will merge after 2 tiny comments fixed :)

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[demos/model-builder/model-builder.html, line 250 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-KuWRjZ66nalPXeOY4mH:-KuWRjZ66nalPXeOY4mI:b-h43bne) ([raw file](https://github.com/pair-code/deeplearnjs/blob/63c8768058c70021edcee0079d1c6d985ba7d5a6/demos/model-builder/model-builder.html#L250)):*
> ```HTML
>       </paper-input>
> 
>       <paper-input no-animations label="Gamma" id="gamma" disabled="[[!datasetDownloaded]]" value={{gamma}}>
> ```

can you condition these paper inputs on the optimizer (gamma doesn't apply to momentum, for example)

---

*[demos/model-builder/model-builder.ts, line 362 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-KuWRgBKZnUopOqjw1kW:-KuWRgBKZnUopOqjw1kX:b5wws46) ([raw file](https://github.com/pair-code/deeplearnjs/blob/63c8768058c70021edcee0079d1c6d985ba7d5a6/demos/model-builder/model-builder.ts#L362)):*
> ```XML
>     // Recreate optimizer with the selected optimizer and hyperparameters.
>     this.optimizer = this.createOptimizer();
>     console.log(this.optimizer);
> ```

remove console.log

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-:-KuWRoJqQYOC7Kn06ePp:b-hdmvna)*

**Comment by nsthorat** [comment]
QQ - which IDE are you using? Small style comment, if you use visual studio code with clang-format command line tool & the vscode clang-format extension the styling will happen for you every time you save!

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[demos/model-builder/model-builder.ts, line 360 at r2](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-KudqjFqYJ_xp20ubI3D:-KudqjFrX8AiSjMITfEr:b64gll3) ([raw file](https://github.com/pair-code/deeplearnjs/blob/e8f201d968d8e7c499cf08f55f339de63df6c5e8/demos/model-builder/model-builder.ts#L360)):*
> ```XML
>     this.resetHyperParamRequirements();
>     switch (optimizerName) {
>     case "sgd": {
> ```

indent the case block by 2 more

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-:-Kudqm7G2tBHcoRbEfSP:byk8yeh)*

**Comment by Lewuathe** [comment]
I use clang-format plugin in Sublime Text 3. After I changed format style to "Google" from "LLVM", it works as expected. 
Thanks

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-:-Kug4NZQKIsAKtJeBHCE:b-9lo0hv)*

**Comment by Lewuathe** [comment]
screen shot 2017-09-23 at 9 22 34
screen shot 2017-09-23 at 9 22 46

**Comment by nsthorat** [comment]
:lgtm_strong:

---

Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/114#-:-KuyBHupZW9rsbR6hNFB:b-jbsf8z)*
